backend/src/services/integrationService.ts

```diff
@@ -155,84 +155,132 @@ export default class IntegrationService {
+      const integration = await IntegrationRepository.create(data, txOptions)
+
+      const collectionService = new CollectionService(txOptions)
 
       const [insightsProject] = await collectionService.findInsightsProjectsBySegmentId(
-        record.segmentId,
+        integration.segmentId,
       )
 
       if (!insightsProject) {
-        return record
+        this.options.log.info(
+          `The segmentId: ${integration.segmentId} does not have any InsightsProject related`,
+        )
+        return integration
       }
 
-      const qx = SequelizeRepository.getQueryExecutor({
-        ...(options || this.options),
-        transaction,
-      })
+      const { segmentId, id: insightsProjectId } = insightsProject
+      const { platform } = data
 
-      if (IntegrationService.isCodePlatform(data.platform)) {
-        const repositories = await collectionService.findRepositoriesForSegment(record.segmentId)
-        data.repositories = [
-          ...new Set([
-            ...Object.values(repositories).flatMap((entries) => entries.map((e) => e.url)),
-          ]),
-        ]
-
-        // TODO: remove the insightsProjectId check when this is not mandatory anymore
-        await insertSegmentRepositories(qx, {
-          insightsProjectId: insightsProject.id,
-          repositories: CollectionService.normalizeRepositories(data.repositories),
-          segmentId: record.segmentId,
-        })
-      }
+      const repositories = IntegrationService.isCodePlatform(platform)
+        ? await this.syncSegmentRepositories({
+            insightsProjectId,
+            integrationId: integration.id,
+            segmentId,
+            txOptions,
+          })
+        : []
 
       await this.updateInsightsProject({
-        insightsProjectId: insightsProject.id,
+        insightsProjectId,
         isFirstUpdate: true,
-        platform: data.platform,
-        segmentId: record.segmentId,
+        platform,
+        repositories,
+        segmentId,
         transaction,
       })
 
-      return record
+      return integration
     } catch (error) {
       SequelizeRepository.handleUniqueFieldError(error, this.options.language, 'integration')
       throw error
     }
   }
 
+  async update(id, data, transaction?: any, options?: IRepositoryOptions) {
+    try {
+      const txOptions = {
+        ...(options || this.options),
+        transaction,
+      }
+
+      const integration = await IntegrationRepository.update(id, data, txOptions)
+
+      const collectionService = new CollectionService(txOptions)
+
+      const [insightsProject] = await collectionService.findInsightsProjectsBySegmentId(
+        integration.segmentId,
+      )
+
+      if (!insightsProject) {
+        this.options.log.info(
+          `The segmentId: ${integration.segmentId} does not have any InsightsProject related`,
+        )
+        return integration
+      }
```

Comment:
I think this part:
```
if (IntegrationService.isCodePlatform(platform) && platform !== PlatformType.GIT) {
        this.gitConnectOrUpdate({
          remotes: repositories,
        })
      }
```
Should be before this code block.
Let's say that we don't have an insights project, we still want the git integration to be updated.
And the way that it's ordered, we wouldn't update the git integration (if I'm seeing this correctly).

Besides this I think everything else makes sense, because everything else is mainly based on the insightsProjectId.